urld0f681c558_farom57_AstroidDriver_tgz-pJ8-laazotea_indi_server_examples_MinimumINDIServerJ8/src/laazotea/indi/INDISexagesimalFormatter.java

```diff
@@ -0,0 +1,355 @@
+   * @throws IllegalArgumentException
+   * @deprecated
+   */
+  @Deprecated
+  public double parseSexagesimal2(String number) throws IllegalArgumentException {
+    number = number.replace(' ', ':');
+    number = number.replace(';', ':');
+
+//    System.out.println(" ->" + number + " ; " + format);
+//    System.out.flush();
+
+    if (number.indexOf(":") == -1) {  // If there are no separators maybe they have sent just a single double
+      try {
+        double n = Double.parseDouble(number);
+
+        return n;
+      } catch (NumberFormatException e) {
+        throw new IllegalArgumentException("Sexagesimal number format not correct (not even single number)");
+      }
+    }
+
+    StringTokenizer st = new StringTokenizer(number, ":", false);
+
+    double degrees = 0;
+    double minutes = 0;
+    double seconds = 0;
+
+    try {
+      String aux = st.nextToken().trim();
+
+      if (aux.length() > 0) {
+        degrees = Double.parseDouble(aux);
+      }
+
+      aux = st.nextToken().trim();
+
+      if (aux.length() > 0) {
+        minutes = Double.parseDouble(aux);
+      }
+
+      if (fractionLength > 5) {
+        aux = st.nextToken().trim();
+
+        if (aux.length() > 0) {
+          seconds = Double.parseDouble(aux);
+        }
+      }
+
+    } catch (NoSuchElementException e) {
+      throw new IllegalArgumentException("Sexagesimal number format not correct");
+    } catch (NumberFormatException e) {
+      throw new IllegalArgumentException("Sexagesimal number component not correct");
+    }
+
+    double res = degrees;
+    if (degrees > 0) {
+      res += (minutes / 60.0) + (seconds / 3600.0);
+    } else {
+      res -= (minutes / 60.0) + (seconds / 3600.0);
+    }
+
+    return res;
+  }
+
+  /**
+   * Parses a sexagesimal number. The input
+   * <code>String</code> is formatted as a maximum of three doubles separated by
+   * : ; or a blank space. The first number represents the number of degrees,
+   * the second is the number of minutes and the third is the number of seconds.
+   *
+   * @param number The number to be parsed.
+   * @return The parsed double.
+   * @throws IllegalArgumentException if the number format is not correct.
+   */
+  public double parseSexagesimal(String number) throws IllegalArgumentException {
+    number = number.trim();
+
+    if (number.length() == 0) {
+      throw new IllegalArgumentException("Empty number");
+    }
+
+    number = number.replace(' ', ':');
+    number = number.replace(';', ':');
+
+    int charCount = number.length() - number.replaceAll(":", "").length();
+
+    if (charCount > 2) {
+      throw new IllegalArgumentException("Too many components for the sexagesimal formatter");
+    }
+
+    double degrees = 0;
+    double minutes = 0;
+    double seconds = 0;
+
+    StringTokenizer st = new StringTokenizer(number, ":", false);
+
+    String d = st.nextToken().trim();
+
+    try {
+      degrees = Double.parseDouble(d);
+    } catch (NumberFormatException e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)
Similar issue at line numbers 218, 231.